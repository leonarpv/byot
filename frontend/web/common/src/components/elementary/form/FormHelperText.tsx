import * as React from 'react';
import {Collapse, FormHelperTextProps, FormHelperText as FormHelperTextBase} from '@material-ui/core';

interface Props extends FormHelperTextProps {
  children?: React.ReactNode;
}

export const FormHelperText: React.FC<Props> = (props: Props) => {
  return (
    <Collapse in={!!props.children}>
      <FormHelperTextBase {...props}>{props.children || ' '}</FormHelperTextBase>
    </Collapse>
  );
};
