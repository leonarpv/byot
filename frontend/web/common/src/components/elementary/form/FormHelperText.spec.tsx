import * as React from 'react';
import {render} from '@testing-library/react';
import {FormHelperText} from './FormHelperText';

describe('<FormHelperText/>', () => {
  it('should render', () => {
    const {container} = render(<FormHelperText />);
    expect(container).toMatchSnapshot();
  });
});
