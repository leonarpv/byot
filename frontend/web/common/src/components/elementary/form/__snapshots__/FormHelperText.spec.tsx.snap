// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormHelperText/> should render 1`] = `
<div>
  <div
    class="MuiCollapse-container MuiCollapse-hidden"
    style="min-height: 0px;"
  >
    <div
      class="MuiCollapse-wrapper"
    >
      <div
        class="MuiCollapse-wrapperInner"
      >
        <p
          class="MuiFormHelperText-root"
        >
          <span>
            â€‹
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;
